messaging/src/main/java/io/lantern/messaging/RecoveryKey.kt

```diff
@@ -0,0 +1,54 @@
+package io.lantern.messaging
+
+import java.security.InvalidKeyException
+import java.security.SecureRandom
+import org.whispersystems.curve25519.Curve25519
+import org.whispersystems.libsignal.ecc.ECKeyPair
+import org.whispersystems.libsignal.ecc.ECPrivateKey
+import org.whispersystems.libsignal.ecc.ECPublicKey
+import org.whispersystems.libsignal.kdf.HKDF
+
+private val hkdf = HKDF.createFor(3)
+
+/**
+ * RecoveryKey is a seed for generating various keys used in the application. Keys are derived
+ * using the HKDF defined in rfc5869.
+ */
+internal typealias RecoveryKey = ByteArray
+
+/**
+ * Generates an ECKeyPair with a derived secret from this RecoveryKey, using the given label.
+ */
+@Throws(InvalidKeyException::class)
+internal fun RecoveryKey.keyPair(label: String): ECKeyPair {
+    val derived = derive(label, 32)
```

Comment:
My only thought here is to make sure to use a string of sufficient length or you don't gain as much by using HKDF

"An important property of HKDF is that it does not amplify entropy but does allow a large source of weaker entropy to be utilized more evenly and effectively."